File: reference/mathexpr/src/token-types.test.ts
import { describe, test, expect } from "bun:test";
import { token } from "./token-types";

describe("token-types", () => {
  test("token creates a Token with kind and value", () => {
    const t = token("number", "42");
    expect(t.kind).toBe("number");
    expect(t.value).toBe("42");
  });

  test("token creates operator tokens", () => {
    expect(token("plus", "+")).toEqual({ kind: "plus", value: "+" });
    expect(token("minus", "-")).toEqual({ kind: "minus", value: "-" });
    expect(token("star", "*")).toEqual({ kind: "star", value: "*" });
    expect(token("slash", "/")).toEqual({ kind: "slash", value: "/" });
    expect(token("percent", "%")).toEqual({ kind: "percent", value: "%" });
    expect(token("power", "**")).toEqual({ kind: "power", value: "**" });
    expect(token("lparen", "(")).toEqual({ kind: "lparen", value: "(" });
    expect(token("rparen", ")")).toEqual({ kind: "rparen", value: ")" });
  });
});

File: research/experiments/sweep/runs/mathexpr-full-kotlin-sonnet45/src/test/kotlin/mathexpr/TokenTypesTest.kt
package mathexpr

import kotlin.test.Test
import kotlin.test.assertEquals

class TokenTypesTest {
    @Test
    fun `creates number token`() {
        val t = token(TokenKind.NUMBER, "42")
        assertEquals(TokenKind.NUMBER, t.kind)
        assertEquals("42", t.value)
    }

    @Test
    fun `creates plus token`() {
        val t = token(TokenKind.PLUS, "+")
        assertEquals(TokenKind.PLUS, t.kind)
        assertEquals("+", t.value)
    }
}
